### ISSUE TITLE
Installing New KIWI version

### ISSUE THREAD

**Issue by NimaDolatabadi** [emolch/kiwi#19]
sorry for making many issues but I believe it is needed.
I have attached a complete list of installed software on my Ubuntu that varies in using `apt-get` or `python`.
_My system properties is shown below:_
**Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial**

And my **Hardware properties** is withing `Hardware-Properties.txt` that I've attached it as well.
I Have installed `HDF5-1.10.5` , `FFTW3-3.3.8` and [LibMseed](https://iris-edu.github.io/libmseed).
I must mention that `lmplatform.h` is no longer supported by `libmseed` you can find out why in here [lmplatform.h](https://github.com/iris-edu/libmseed/issues/16#issuecomment-339721391), so I foound this **file** in one of `seisan` directories.
Now I'm trying to compile `KIWI` but now I'm getting the following error which shows that `mseed_simple.o` cannot be complied. I have attached **Makefile.local** as well.

When I simply type `make` it gives me the following errors.

make -C mseed/ && touch .mseedsimple
make[1]: Entering directory '/home/nima/src/kiwi/mseed'
gcc -c -std=c99 -Wall  mseed_simple.c
mseed_simple.c: In function ‘writemseed’:
mseed_simple.c:15:5: error: unknown type name ‘MSTrace’
     MSTrace     *trace;
     ^
mseed_simple.c:20:5: error: unknown type name ‘hptime_t’
     hptime_t    hptime;
     ^
mseed_simple.c:28:13: warning: implicit declaration of function ‘mst_init’ [-Wimplicit-function-declaration]
     trace = mst_init (NULL);
             ^
mseed_simple.c:28:11: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
     trace = mst_init (NULL);
           ^
mseed_simple.c:30:19: error: request for member ‘network’ in something not a structure or union
     strncpy( trace->network, network, 2 );
                   ^
mseed_simple.c:31:19: error: request for member ‘station’ in something not a structure or union
     strncpy( trace->station, station, 4 );
                   ^
mseed_simple.c:32:19: error: request for member ‘location’ in something not a structure or union
     strncpy( trace->location, location, 2 );
                   ^
mseed_simple.c:33:19: error: request for member ‘channel’ in something not a structure or union
     strncpy( trace->channel, channel, 3 );
                   ^
mseed_simple.c:34:10: error: request for member ‘network’ in something not a structure or union
     trace->network[3]  = '\0';
          ^
mseed_simple.c:35:10: error: request for member ‘station’ in something not a structure or union
     trace->station[5]  = '\0';
          ^
mseed_simple.c:36:10: error: request for member ‘location’ in something not a structure or union
     trace->location[3] = '\0';
          ^
mseed_simple.c:37:10: error: request for member ‘channel’ in something not a structure or union
     trace->channel[4]  = '\0';
          ^
mseed_simple.c:45:15: error: ‘hptime_t’ undeclared (first use in this function)
     hptime = (hptime_t)(sec_em5 * HPTMODULUS);
               ^
mseed_simple.c:45:15: note: each undeclared identifier is reported only once for each function it appears in
mseed_simple.c:45:35: error: ‘HPTMODULUS’ undeclared (first use in this function)
     hptime = (hptime_t)(sec_em5 * HPTMODULUS);
                                   ^
mseed_simple.c:47:10: error: request for member ‘starttime’ in something not a structure or union
     trace->starttime = hptime;
          ^
mseed_simple.c:48:10: error: request for member ‘samprate’ in something not a structure or union
     trace->samprate = 1./deltat;
          ^
mseed_simple.c:49:10: error: request for member ‘sampletype’ in something not a structure or union
     trace->sampletype = 'f';
          ^
mseed_simple.c:55:10: error: request for member ‘datasamples’ in something not a structure or union
     trace->datasamples = seismogram;
          ^
mseed_simple.c:56:10: error: request for member ‘numsamples’ in something not a structure or union
     trace->numsamples = length;
          ^
mseed_simple.c:57:10: error: request for member ‘samplecnt’ in something not a structure or union
     trace->samplecnt = length;
          ^
mseed_simple.c:58:10: error: request for member ‘sampletype’ in something not a structure or union
     trace->sampletype = 'f';
          ^
mseed_simple.c:59:5: warning: implicit declaration of function ‘mst_srcname’ [-Wimplicit-function-declaration]
     mst_srcname (trace, srcname, 0);
     ^
mseed_simple.c:61:16: warning: implicit declaration of function ‘mst_pack’ [-Wimplicit-function-declaration]
     precords = mst_pack (trace, &record_handler, outfile, 4096, DE_FLOAT32,
                ^
mseed_simple.c:64:5: warning: implicit declaration of function ‘mst_free’ [-Wimplicit-function-declaration]
     mst_free( &trace );                                 
     ^
mseed_simple.c:17:27: warning: variable ‘precords’ set but not used [-Wunused-but-set-variable]
     int         psamples, precords;
                           ^
mseed_simple.c: In function ‘readmseed’:
mseed_simple.c:72:5: error: unknown type name ‘MSTraceGroup’
     MSTraceGroup *mstg = NULL;
     ^
mseed_simple.c:73:5: error: unknown type name ‘MSTrace’
     MSTrace      *trace;
     ^
mseed_simple.c:84:5: error: unknown type name ‘hptime_t’
     hptime_t hptime;
     ^
mseed_simple.c:88:15: warning: implicit declaration of function ‘ms_readtraces’ [-Wimplicit-function-declaration]
     retcode = ms_readtraces (&mstg, filename, 0, -1.0, -1.0, 0, 1, 1, 0);
               ^
mseed_simple.c:97:17: error: request for member ‘traces’ in something not a structure or union
     trace = mstg->traces;
                 ^
mseed_simple.c:99:14: error: request for member ‘datasamples’ in something not a structure or union
     if (trace->datasamples == NULL) {
              ^
mseed_simple.c:104:48: error: request for member ‘numsamples’ in something not a structure or union
     data = (float*) calloc(sizeof(float), trace->numsamples);
                                                ^
mseed_simple.c:106:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'i') {
              ^
mseed_simple.c:107:33: error: request for member ‘datasamples’ in something not a structure or union
         source_int = (int*)trace->datasamples;
                                 ^
mseed_simple.c:108:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:112:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'a') {
              ^
mseed_simple.c:113:35: error: request for member ‘datasamples’ in something not a structure or union
         source_char = (char*)trace->datasamples;
                                   ^
mseed_simple.c:114:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:118:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'f') {
              ^
mseed_simple.c:119:37: error: request for member ‘datasamples’ in something not a structure or union
         source_float = (float*)trace->datasamples;
                                     ^
mseed_simple.c:120:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:124:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'd') {
              ^
mseed_simple.c:125:39: error: request for member ‘datasamples’ in something not a structure or union
         source_double = (double*)trace->datasamples;
                                       ^
mseed_simple.c:126:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:131:19: error: request for member ‘starttime’ in something not a structure or union
     hptime = trace->starttime ;
                   ^
mseed_simple.c:134:39: error: ‘HPTMODULUS’ undeclared (first use in this function)
     dtime = (double)hptime / ((double)HPTMODULUS);
                                       ^
mseed_simple.c:136:20: error: request for member ‘numsamples’ in something not a structure or union
     *length = trace->numsamples;
                    ^
mseed_simple.c:138:23: error: request for member ‘samprate’ in something not a structure or union
     *deltat = 1./trace->samprate;
                       ^
mseed_simple.c:141:5: warning: implicit declaration of function ‘mst_freegroup’ [-Wimplicit-function-declaration]
     mst_freegroup (&mstg);
     ^
Makefile:5: recipe for target 'mseed_simple.o' failed
make[1]: *** [mseed_simple.o] Error 1
make[1]: Leaving directory '/home/nima/src/kiwi/mseed'
Makefile:121: recipe for target '.mseedsimple' failed
make: *** [.mseedsimple] Error 2

I wonder how this problem could be solved.

I suggest that it would be better if you put a pre-complied of this software in a branch so In the times of need just by installing dependencies and giving a path, the sotfware would be in touch with lower issues.

[Makefile.local.txt](https://github.com/emolch/kiwi/files/3795908/Makefile.local.txt)
[Hardware-Properties.txt](https://github.com/emolch/kiwi/files/3795863/Hardware-Properties.txt)
[python3-list.txt](https://github.com/emolch/kiwi/files/3795854/python3-list.txt)
[installed-softwares.txt](https://github.com/emolch/kiwi/files/3795855/installed-softwares.txt)

Thank you for devoting you time.
Great Regards

**Comment by emolch** [comment]
Hi Nima,

try to compile it against libmseed version 2.19.

Best regards
Sebastian

**Comment by NimaDolatabadi** [comment]
Dear @emolch  Your solution did actually solve half of my problem. I still had some issues in which `-lmseed` was not found. so in line **number 40** of `Makefile` I have changed `-lmseed` to `-L/home/nima/src/LIBMSEED/lib -lmseed` so that it has been solved completely.
the only thing that matters is that the older version of `libmseed` is hardly accessible but I it from 3 years ago so that did. I believe you've got to edit your information about installing KIWI.
Thank you for devoting your time on this issue.
